### ISSUE TITLE
EasyEffects not working on Manjaro 21.1.1

### ISSUE THREAD

**Issue by tiagaoalb** [wwmm/easyeffects#1130]
![EasyEffects on Manjaro](https://i.imgur.com/PmbkxHN.png)
The app doesnt show anything.

**Comment by vchernin** [comment]
Is this what happens when you play audio through any other app?

**Comment by tiagaoalb** [comment]
> Is this what happens when you play audio through any other app?

Happens when I listen to music through Brave browser.

**Comment by tiagaoalb** [comment]
![Pipewire Screen](https://i.imgur.com/jSbfUEi.png)

**Comment by wwmm** [comment]
Kill the current instance `easyeffects -q` and restart it in debug mode `G_MESSAGES_DEBUG=easyeffects easyeffects`. Maybe some kind of error is in the logs.

**Comment by wwmm** [comment]
Are other apps like mpv, vlc, firefox working?

**Comment by tiagaoalb** [comment]
> Are other apps like mpv, vlc, firefox working?

Nope.

**Comment by tiagaoalb** [comment]
> Kill the current instance `easyeffects -q` and restart it in debug mode `G_MESSAGES_DEBUG=easyeffects easyeffects`. Maybe some kind of error is in the logs.

It showed a lot of messages. Is there a specific one?

**Comment by wwmm** [comment]
>It showed a lot of messages. Is there a specific one?

Show of them.

**Comment by tiagaoalb** [comment]
> > It showed a lot of messages. Is there a specific one?
> 
> Show of them.

(process:31053): easyeffects-DEBUG: 19:06:46.899: main: locale directory: /usr/share/locale
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: system input presets directory: "/etc/easyeffects/input"; 
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: system input presets directory: "/etc/xdg/easyeffects/input"; 
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: system output presets directory: "/etc/easyeffects/output"; 
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: system output presets directory: "/etc/xdg/easyeffects/output"; 
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: user presets directory already exists: /home/tiagaoalb/.config/easyeffects/
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: user presets directory already exists: /home/tiagaoalb/.config/easyeffects/input
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: user presets directory already exists: /home/tiagaoalb/.config/easyeffects/output
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: user presets directory already exists: /home/tiagaoalb/.config/easyeffects/autoload/input
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.952: presets_manager: user presets directory already exists: /home/tiagaoalb/.config/easyeffects/autoload/output

(easyeffects:31053): Gtk-WARNING **: 19:06:46.958: Theme parsing error: main-dark.css:2060:55-56: The color must be the last parameter

(easyeffects:31053): Gtk-WARNING **: 19:06:46.958: Theme parsing error: main-dark.css:2068:55-56: The color must be the last parameter

(easyeffects:31053): Gtk-WARNING **: 19:06:46.959: Theme parsing error: main-dark.css:4178:31-32: Expected shadow value to contain a length

(easyeffects:31053): Gtk-WARNING **: 19:06:46.959: Theme parsing error: main-dark.css:4319:1-2: Expected a valid selector

(easyeffects:31053): Gtk-WARNING **: 19:06:46.960: Theme parsing error: main-dark.css:4871:21-27: Not a valid image

(easyeffects:31053): Gtk-WARNING **: 19:06:46.960: Theme parsing error: main-dark.css:4880:3-15: Junk at end of value for outline-color

(easyeffects:31053): Gtk-WARNING **: 19:06:46.961: Theme parsing error: main-dark.css:5897:1-2: Expected a valid selector

(easyeffects:31053): Gtk-WARNING **: 19:06:46.961: Theme parsing error: main-dark.css:6001:10-20: "text_color" is not a valid color name.

(easyeffects:31053): Gtk-WARNING **: 19:06:46.961: Theme parsing error: main-dark.css:6013:27-45: "sidebar_text_color" is not a valid color name.

(easyeffects:31053): Gtk-WARNING **: 19:06:46.961: Theme parsing error: main-dark.css:6018:10-28: "sidebar_text_color" is not a valid color name.

(easyeffects:31053): Gtk-WARNING **: 19:06:46.961: Theme parsing error: main-dark.css:6108:1-2: Expected a valid selector

(easyeffects:31053): Gtk-WARNING **: 19:06:46.961: Theme parsing error: main-dark.css:6745:1-2: Expected a valid selector
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.968: application: easyeffects version: 6.1.0
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.969: pipe_manager: compiled with pipewire: 0.3.33
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.969: pipe_manager: linked to pipewire: 0.3.34
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.971: pipe_manager: core version: 0.3.33
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.971: pipe_manager: core name: pipewire-0
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.971: pipe_manager: found metadata: settings
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.971: pipe_manager: Audio/Sink 80 easyeffects_sink was added
(easyeffects:31053): easyeffects-DEBUG: 19:06:46.971: pipe_manager: Audio/Source/Virtual 81 easyeffects_source was added
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.009: lv2_wrapper: http://lsp-plug.in/plugins/lv2/sc_compressor_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.034: lv2_wrapper: http://lsp-plug.in/plugins/lv2/comp_delay_x2_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.071: lv2_wrapper: http://lsp-plug.in/plugins/lv2/para_equalizer_x32_lr requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.123: lv2_wrapper: http://lsp-plug.in/plugins/lv2/limiter_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.135: lv2_wrapper: http://lsp-plug.in/plugins/lv2/loud_comp_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.147: lv2_wrapper: urn:zamaudio:ZaMaximX2 requires feature: http://lv2plug.in/ns/ext/options#options
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.147: lv2_wrapper: urn:zamaudio:ZaMaximX2 requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.166: lv2_wrapper: http://lsp-plug.in/plugins/lv2/mb_compressor_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.209: pipe_manager: Filter pe_soe_output_level, id = 42, was added
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.211: soe: output_level successfully connected to pipewire graph
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.211: pipe_manager: Filter pe_soe_spectrum, id = 52, was added
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.214: soe: spectrum successfully connected to pipewire graph
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.214: pipe_manager: Filter pe_soe_bass_enhancer, id = 57, was added
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.217: soe: bass_enhancer successfully connected to pipewire graph

(easyeffects:31053): easyeffects-WARNING **: 19:06:47.218: soe:  link from node 42 to output device 4294967295 failed
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.255: lv2_wrapper: http://lsp-plug.in/plugins/lv2/sc_compressor_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.280: lv2_wrapper: http://lsp-plug.in/plugins/lv2/comp_delay_x2_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.316: lv2_wrapper: http://lsp-plug.in/plugins/lv2/para_equalizer_x32_lr requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.369: lv2_wrapper: http://lsp-plug.in/plugins/lv2/limiter_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.381: lv2_wrapper: http://lsp-plug.in/plugins/lv2/loud_comp_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.393: lv2_wrapper: urn:zamaudio:ZaMaximX2 requires feature: http://lv2plug.in/ns/ext/options#options
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.393: lv2_wrapper: urn:zamaudio:ZaMaximX2 requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.412: lv2_wrapper: http://lsp-plug.in/plugins/lv2/mb_compressor_stereo requires feature: http://lv2plug.in/ns/ext/urid#map
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.453: pipe_manager: Filter pe_sie_output_level, id = 72, was added
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.455: sie: output_level successfully connected to pipewire graph
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.456: pipe_manager: Filter pe_sie_spectrum, id = 77, was added
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.458: sie: spectrum successfully connected to pipewire graph
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.458: sie: Input device id is invalid. Aborting the link between filters in the microphone pipeline
easyeffects-INFO: 19:06:47.458: application: disabling global bypass

(easyeffects:31053): easyeffects-WARNING **: 19:06:47.460: soe:  link from node 42 to output device 4294967295 failed
(easyeffects:31053): easyeffects-DEBUG: 19:06:47.460: sie: Input device id is invalid. Aborting the link between filters in the microphone pipeline

(easyeffects:31053): Gtk-WARNING **: 19:06:47.533: Theme parsing error: main-dark.css:2060:55-56: The color must be the last parameter

(easyeffects:31053): Gtk-WARNING **: 19:06:47.533: Theme parsing error: main-dark.css:2068:55-56: The color must be the last parameter

(easyeffects:31053): Gtk-WARNING **: 19:06:47.535: Theme parsing error: main-dark.css:4178:31-32: Expected shadow value to contain a length

(easyeffects:31053): Gtk-WARNING **: 19:06:47.535: Theme parsing error: main-dark.css:4319:1-2: Expected a valid selector

(easyeffects:31053): Gtk-WARNING **: 19:06:47.535: Theme parsing error: main-dark.css:4871:21-27: Not a valid image

(easyeffects:31053): Gtk-WARNING **: 19:06:47.535: Theme parsing error: main-dark.css:4880:3-15: Junk at end of value for outline-color

(easyeffects:31053): Gtk-WARNING **: 19:06:47.536: Theme parsing error: main-dark.css:5897:1-2: Expected a valid selector

(easyeffects:31053): Gtk-WARNING **: 19:06:47.536: Theme parsing error: main-dark.css:6001:10-20: "text_color" is not a valid color name.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.536: Theme parsing error: main-dark.css:6013:27-45: "sidebar_text_color" is not a valid color name.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.536: Theme parsing error: main-dark.css:6018:10-28: "sidebar_text_color" is not a valid color name.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.536: Theme parsing error: main-dark.css:6108:1-2: Expected a valid selector

(easyeffects:31053): Gtk-WARNING **: 19:06:47.536: Theme parsing error: main-dark.css:6745:1-2: Expected a valid selector

(easyeffects:31053): Gtk-WARNING **: 19:06:47.600: GtkImage 0x5611adb6b300 reported baselines of minimum -2147483648 and natural -2147483648, but sizes of minimum 16 and natural 16. Baselines must be inside the widget size.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.600: GtkImage 0x5611adb6b620 reported baselines of minimum -2147483648 and natural -2147483648, but sizes of minimum 16 and natural 16. Baselines must be inside the widget size.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.600: GtkImage 0x5611adb6b940 reported baselines of minimum -2147483648 and natural -2147483648, but sizes of minimum 16 and natural 16. Baselines must be inside the widget size.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.606: GtkImage 0x5611adb6b300 reported baselines of minimum -2147483648 and natural -2147483648, but sizes of minimum 16 and natural 16. Baselines must be inside the widget size.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.606: GtkImage 0x5611adb6b620 reported baselines of minimum -2147483648 and natural -2147483648, but sizes of minimum 16 and natural 16. Baselines must be inside the widget size.

(easyeffects:31053): Gtk-WARNING **: 19:06:47.606: GtkImage 0x5611adb6b940 reported baselines of minimum -2147483648 and natural -2147483648, but sizes of minimum 16 and natural 16. Baselines must be inside the widget size.

**Comment by tiagaoalb** [comment]
> > It showed a lot of messages. Is there a specific one?
> 
> Show of them.

Acabei de ver no seu perfil que é BR 😃

**Comment by wwmm** [comment]
```
easyeffects-DEBUG: 19:06:46.969: pipe_manager: compiled with pipewire: 0.3.33
easyeffects-DEBUG: 19:06:46.969: pipe_manager: linked to pipewire: 0.3.34
easyeffects-DEBUG: 19:06:46.971: pipe_manager: core version: 0.3.33
```
Humm... I wonder if this is one of those cases where the package providing EasyEffects should be recompiled after an upgrade to the PipeWire package. Or maybe it would be the case of rebooting the computer. Sometimes users have bizarre issues after upgrading PipeWire or Pulseaudio and a reboot fixes it.

>easyeffects-WARNING **: 19:06:47.460: soe: link from node 42 to output device 4294967295 failed

We are not being able to get the the output device. And id like `4294967295` is a sign of that. It seems to me we are having problems talking to the server. Did you do any upgrade recently?

**Comment by tiagaoalb** [comment]
> ```
> easyeffects-DEBUG: 19:06:46.969: pipe_manager: compiled with pipewire: 0.3.33
> easyeffects-DEBUG: 19:06:46.969: pipe_manager: linked to pipewire: 0.3.34
> easyeffects-DEBUG: 19:06:46.971: pipe_manager: core version: 0.3.33
> ```
> 
> Humm... I wonder if this is one of those cases where the package providing EasyEffects should be recompiled after an upgrade to the PipeWire package. Or maybe it would be the case of rebooting the computer. Sometimes users have bizarre issues after upgrading PipeWire or Pulseaudio and a reboot fixes it.
> 
> > easyeffects-WARNING **: 19:06:47.460: soe: link from node 42 to output device 4294967295 failed
> 
> We are not being able to get the the output device. And id like `4294967295` is a sign of that. It seems to me we are having problems talking to the server. Did you do any upgrade recently?

This is a clean install of Manjaro and I already rebooted my PC.

**Comment by wwmm** [comment]
>This is a clean install of Manjaro and I already rebooted my PC.

Then it is more likely Manjaro package maintainers forgot to recompile our package after upgrading PipeWire. The difference in version shown in the logs may be a sign of that. I use Arch Linux and things are fine. Something you can try is installing EasyEffects through Flatpak. If it works it is definitely an issue with the Manjaro package.

**Comment by tiagaoalb** [comment]
> > This is a clean install of Manjaro and I already rebooted my PC.
> 
> Then it is more likely Manjaro package maintainers forgot to recompile our package after upgrading PipeWire. The difference in version shown in the logs may be a sign of that. I use Arch Linux and things are fine. Something you can try is installing EasyEffects through Flatpak. If it works it is definitely an issue with the Manjaro package.

Nice, I'll do that 👊🏻

**Comment by wwmm** [comment]
I haven't tried to install a Flatpak package in years. @vchernin what would be the fast way to try it?

**Comment by vchernin** [comment]
> ![PipeWire Screen](https://i.imgur.com/jSbfUEi.png)

@wwmm look at the PipeWire versions here? does this make sense (given what you discovered recently about what PipeWire version reporting)

I think it's defintely a PipeWire version mismatch issue.

If you've never installed a Flatpak before [follow this](https://flatpak.org/setup/). Once that's done [just follow the EasyEffects instructions here](https://flathub.org/apps/details/com.github.wwmm.easyeffects).

**Comment by tiagaoalb** [comment]
Just installed from flatpak and this is the result.

![EasyEffect from Flatpak](https://i.imgur.com/Ol09Hd1.png)

**Comment by tiagaoalb** [comment]
> > ![PipeWire Screen](https://camo.githubusercontent.com/e13c0689c13001d45eece3ae108c8cf5ed4591b51711c9b9de2c368f34b5d24b/68747470733a2f2f692e696d6775722e636f6d2f6a5362665545692e706e67)
> 
> @wwmm look at the PipeWire versions here does this make sense? (given what you discovered recently about PipeWire version reporting)
> 
> I think it's defintely a PipeWire version mismatch issue.
> 
> If you've never installed a Flatpak before [follow this](https://flatpak.org/setup/). Once that's done [just follow the EasyEffects instructions here](https://flathub.org/apps/details/com.github.wwmm.easyeffects) (it's the same link as what's in the readme).

Installed from flatpak and the result is the same...

**Comment by vchernin** [comment]
> Just installed from flatpak and this is the result.
> 
> 
> 
> ![EasyEffect from Flatpak](https://i.imgur.com/Ol09Hd1.png)

So apps playing audio still don't show up then right?

Can you try and see what pacman is doing on Manjaro? Can you try to update the pipewire package and see what happens?

**Comment by tiagaoalb** [comment]
> > Just installed from flatpak and this is the result.
> > ![EasyEffect from Flatpak](https://camo.githubusercontent.com/52a83dd1680a1dcef43d33aa050360347fc6cdf9e85deac60984484127b9440d/68747470733a2f2f692e696d6775722e636f6d2f4f6c30394864312e706e67)
> 
> So apps playing audio still don't show up then right?
> 
> Can you try and see what pacman is doing on Manjaro? Can you try to update the pipewire package and see what happens?

pactl info returned this:

Server String: /run/user/1000/pulse/native
Library Protocol Version: 35
Server Protocol Version: 35
Is Local: yes
Client Index: 12
Tile Size: 65472
User Name: tiagaoalb
Host Name: tiago-manjaro
Server Name: pulseaudio
Server Version: 15.0
Default Sample Specification: s16le 2ch 44100Hz
Default Channel Map: front-left,front-right
Default Sink: alsa_output.pci-0000_05_04.0.iec958-stereo
Default Source: alsa_input.pci-0000_00_1f.3.analog-stereo
Cookie: 4147:e332

**Comment by vchernin** [comment]
> 
> 
> 
> 
> 
> 
> > > Just installed from flatpak and this is the result.
> 
> > > ![EasyEffect from Flatpak](https://camo.githubusercontent.com/52a83dd1680a1dcef43d33aa050360347fc6cdf9e85deac60984484127b9440d/68747470733a2f2f692e696d6775722e636f6d2f4f6c30394864312e706e67)
> 
> > 
> 
> > So apps playing audio still don't show up then right?
> 
> > 
> 
> > Can you try and see what pacman is doing on Manjaro? Can you try to update the pipewire package and see what happens?
> 
> 
> 
> pactl info returned this:
> 
> 
> 
> Server String: /run/user/1000/pulse/native
> 
> Library Protocol Version: 35
> 
> Server Protocol Version: 35
> 
> Is Local: yes
> 
> Client Index: 12
> 
> Tile Size: 65472
> 
> User Name: tiagaoalb
> 
> Host Name: tiago-manjaro
> 
> Server Name: pulseaudio
> 
> Server Version: 15.0
> 
> Default Sample Specification: s16le 2ch 44100Hz
> 
> Default Channel Map: front-left,front-right
> 
> Default Sink: alsa_output.pci-0000_05_04.0.iec958-stereo
> 
> Default Source: alsa_input.pci-0000_00_1f.3.analog-stereo
> 
> Cookie: 4147:e332
> 
> 


At least for me on Fedora I don't have the `pactl` command. Are you confident pipewire is actually installed, and that you've replace pulseaudio with pipewire-pulse. Otherwise brave and other apps will likely use the pulseaudio backend, while EasyEffects uses the pipewire backend that's not connected to the pulseaudio apps.

The easiest way to make sure is by verifying `pipewire-pulse` is installed.

**Comment by tiagaoalb** [comment]
> > > > Just installed from flatpak and this is the result.
> > 
> > 
> > > > ![EasyEffect from Flatpak](https://camo.githubusercontent.com/52a83dd1680a1dcef43d33aa050360347fc6cdf9e85deac60984484127b9440d/68747470733a2f2f692e696d6775722e636f6d2f4f6c30394864312e706e67)
> > 
> > 
> > > 
> > 
> > 
> > > So apps playing audio still don't show up then right?
> > 
> > 
> > > 
> > 
> > 
> > > Can you try and see what pacman is doing on Manjaro? Can you try to update the pipewire package and see what happens?
> > 
> > 
> > pactl info returned this:
> > Server String: /run/user/1000/pulse/native
> > Library Protocol Version: 35
> > Server Protocol Version: 35
> > Is Local: yes
> > Client Index: 12
> > Tile Size: 65472
> > User Name: tiagaoalb
> > Host Name: tiago-manjaro
> > Server Name: pulseaudio
> > Server Version: 15.0
> > Default Sample Specification: s16le 2ch 44100Hz
> > Default Channel Map: front-left,front-right
> > Default Sink: alsa_output.pci-0000_05_04.0.iec958-stereo
> > Default Source: alsa_input.pci-0000_00_1f.3.analog-stereo
> > Cookie: 4147:e332
> 
> At least for me on Fedora I don't have the `pactl` command. Are you confident pipewire is actually installed, and that you've replace pulseaudio with pipewire-pulse. Otherwise brave and other apps will likely use the pulseaudio backend, while EasyEffects uses the pipewire backend that's not connected to the pulseaudio apps.
> 
> The easiest way to make sure is by verifying `pipewire-pulse` is installed.

This package is not installed, but pulseaudio is installed and Manjaro is using it.
Is there a solution for this?

**Comment by vchernin** [comment]
Yes, just install pipewire-pulse with pacman. It will tell you it conflicts with pulseaudio, simply agree with it and remove pulseaudio in favor of pipewire-pulse.

**Comment by vchernin** [comment]
And to add on, if you have JACK or ALSA apps you will want pipewire-alsa and pipewire-jack as well.

**Comment by tiagaoalb** [comment]
> Yes, just install pipewire-pulse with pacman. It will tell you it conflicts with pulseaudio, simply agree with it and it will remove pulseaudio in favor of pipewire-pulse. Maybe reboot afterwards.

sudo pacman -S pipewire-pulse
resolving dependencies...
looking for conflicting packages...
:: pipewire-pulse and pulseaudio are in conflict. Remove pulseaudio? [y/N] y
:: pipewire-pulse and pulseaudio-bluetooth are in conflict. Remove pulseaudio-bluetooth? [y/N] y
error: failed to prepare transaction (could not satisfy dependencies)
:: removing pulseaudio breaks dependency 'pulseaudio=15.0-1' required by pulseaudio-equalizer
:: removing pulseaudio breaks dependency 'pulseaudio=15.0-1' required by pulseaudio-jack
:: removing pulseaudio breaks dependency 'pulseaudio=15.0-1' required by pulseaudio-lirc
:: removing pulseaudio breaks dependency 'pulseaudio=15.0-1' required by pulseaudio-rtp
:: removing pulseaudio breaks dependency 'pulseaudio=15.0-1' required by pulseaudio-zeroconf

**Comment by vchernin** [comment]
Well you won't be able to proceed to fully switch to PipeWire unless you give up pulseaudio-equalizer and those other packages. Do you need those? You can always get them back if you want to switch back to pulseaudio.

If you don't need those just remove them manually with pacman.

Oh and I have no idea what it's for, but I doubt manjaro-pulse is important. But I couldn't find any info about it ¯\\\_(ツ)_/¯

**Comment by tiagaoalb** [comment]
> Well you won't be able to proceed to fully switch to PipeWire unless you give up pulseaudio-equalizer and those other packages. Do you need those? You can always get them back if you want to switch back to pulseaudio.
> 
> If you don't need those just remove them manually with pacman.
> 
> Oh and I have no idea what it's for, but I doubt manjaro-pulse is important. But I couldn't find any info about it ¯\_(ツ)_/¯

It worked. Thank you very much for your patience and attention 👊🏻

**Comment by vchernin** [comment]
> It worked. Thank you very much for your patience and attention 👊🏻

Nice, so are both the Arch package and Flatpak working as intended now?

**Comment by tiagaoalb** [comment]
> > It worked. Thank you very much for your patience and attention 👊🏻
> 
> Nice, so are both the Arch package and Flatpak working as intended now?

Yes, I tried them both, and they are working fine.

**Comment by vchernin** [comment]
@wwmm so the problem here wasn't really EasyEffects fault, but other apps not connecting to PipeWire and EasyEffects therefore having no idea about them.

As far as I can tell the only practical improvement would be for the Arch EasyEffects package to have a hard dependency on pipewire-pulse, pipewire-alsa and pipewire-jack. So EasyEffects could not be installed without first making sure all most packages of pulseaudio and others are removed and replaced with pipewire versions.

But I think some users will not want to install such packages, and it's too easy for users simply not see what optional depedencies a package might have. So I don't know if there's anything better we can do.

I also have basically no ideas what could be done for the Flatpak. The Flatpak by nature already has limited access to the system. The only solution there would be to get EasyEffects Flatpak to be able to require certain packages on the host, which isn't really possible currently. I might ask anyways as it still seems like useful functionality.

We could also poke holes in the Flatpak sandbox to sniff around and find evidence of pulseaudio installed, but that also isn't great.

**Comment by wwmm** [comment]
>As far as I can tell the only practical improvement would be for the Arch EasyEffects package to have a hard dependency on pipewire-pulse, pipewire-alsa and pipewire-jack.

For most cases only `pipewire-pulse` is really necessary. It provides PipeWire's compatibility layer for Pulseaudio applications. 

>But I think some users will not want to install such packages

That is right. There are situations where the user may have PipeWire installed only to make screen sharing working on Wayland. So forcing pipewire-pulse installation is something that has to be done carefully. I am not sure how this could be solved at this moment. Even if I say that pipewire-pulse is required package maintainers may decide to not force its installation for other reasons. 

Whatever solution is done to this it will have to be done on the package management level. I do not see much we can do from here.

**Comment by Souravgoswami** [comment]
I have faced the same issue on Arch linux for months, and I just kept using pulseeffects (legacy version from AUR). But today pulseeffects gave up, and it's not working anymore. Tried easyeffects and it didn't work. I have solved the issue. Just commenting in case someone needs the help.

pipewire-pulse is compatible with pulseaudio. There were some of the dependent packages for me. I uninstalled them because I don't need those. Replaced pulseaudio with pipewire-pulse, and things are working great. Easyeffects is also working great!
